fix(relay): Update relay configuration (#7062)
getsentry/sentry-docs: src/docs/product/relay/options.mdx

  - /meta/relay/options/
---

Configuration for Relay is recorded either in the file `.relay/config.yml`, or when noted, can be overridden through an environment variable. To change the config location, pass the `--config` option to any Relay command:

```shell
❯ ./relay run --config /path/to/folder
```

All configuration keys are `snake_case`.

## Credentials

Credentials can be initiated as described in the [getting started page](/product/relay/getting-started/), but can be overridden by setting the following environment variables:

> - secret key: `"RELAY_SECRET_KEY"`
> - public key: `"RELAY_PUBLIC_KEY"`
> - id: `"RELAY_ID"`

Reviewer: Starting the sentences with uppercase look better to me, and I don't think we need the `"`s:
```suggestion
- Secret key: `RELAY_SECRET_KEY`.
- Public key: `RELAY_PUBLIC_KEY`.
- ID: `RELAY_ID`.
```

> Credentials can be initiated as described in the [getting started page](https://sentry-docs-git-relayconfig.sentry.dev/product/relay/getting-started/?original_referrer=https%3A%2F%2Fgithub.com%2Fgetsentry%2Fsentry-docs%2Fpull%2F7062#creating-credentials), but can be overriden by setting the following environment variables:

Reviewer: I don't know what's the best practice for links here, but you don't _need_ to write the full URL, see:
https://github.com/getsentry/sentry-docs/blob/55c4d20aaded7449ac21ec792f615a541952293e/src/platforms/android/using-ndk.mdx?plain=1#L19
I think we can get rid of the referrer parameter too?
@lizokm may be able to provide some guidance here.

> Credentials can be initiated as described in the [getting started page](https://sentry-docs-git-relayconfig.sentry.dev/product/relay/getting-started/?original_referrer=https%3A%2F%2Fgithub.com%2Fgetsentry%2Fsentry-docs%2Fpull%2F7062#creating-credentials), but can be overriden by setting the following environment variables:

Reviewer 2: ```suggestion
Credentials can be initiated as described in the [getting started page](/product/relay/getting-started/), but can be overridden by setting the following environment variables:
```
:)

> id: `"RELAY_ID"`

Reviewer: This renders in a row. What do you think about creating an actual list for this?

Reviewer 2: I think a list will be easier to quickly read :)

Author: i made it now an unordered list

## Relay

| key                        | type    | value                                                                                         | description                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| -------------------------- | ------- | --------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | relay.mode                 | String  | default value: `managed` possible values: `managed, static, proxy, capture` optional: `false` | Controls how Relay obtains the project configuration for events. For detailed explanation of these modes, see [Relay Modes](/product/relay/modes/). `Environment variable: "RELAY_MODE."`                                                                                                                                                                                                                                                       |

Reviewer: > optional: `false`

Seems confusing to me. What about `optional: no`? Typing the backticks suggests you can set `false`.

> | relay.mode                 | String  | default value: `managed` possible values: `managed, static, proxy, capture` optional: `false` | Controls how Relay obtains the project configuration for events. For detailed explanation of these modes, see [Relay Modes](/product/relay/modes/). `Environment variable: "RELAY_MODE."`                                                                                                                                                                                                                                                       |

Reviewer: If we're creating a table, I think it'd be less confusing if the environment variable is another column like `key`, instead of a note in the description. See how we do with [Next.js docs](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#use-configuration-files-for-source-map-upload), as an example.

Reviewer 2: Ah yes, I tend to agree. I think the more we can split out the information the easier it will be to read <3

Author: this was my plan, but with 5 columns it will crash with the "ON THIS PAGE" index, is there a way to go around this? @lizokm

Reviewer 2: @TBS1996 fixing the overlap of the "on this page" index is in our backlog to figure out, but we haven't fixed it yet, unfortunately. So I think we should just leave it as 4 columns for now and then we don't have to worry about the Size Limits table having one lonely entry :) 

In any case, I think this change will make the page much easier to read even if we don't have a separate column for environment variables!

  - /meta/relay/options/
---

> Configuration for Relay is recorded either in the file `.relay/config.yml`, or when noted, will be overridden through an environment variable. To change the config location, pass the `--config` option to any Relay command:

Author: @jan-auer  suggested that the docs should list the options that can be chosen from environment variables. Since this page doesn't show all the relay config options (by design i assume), I've only noted on the current shown options where they can be also configured by env-vars. 

I'm looking for feedback on whether we should have a separate page for all the environmental variables regardless, or keep it like this.

> Configuration for Relay is recorded either in the file `.relay/config.yml`, or when noted, will be overridden through an environment variable. To change the config location, pass the `--config` option to any Relay command:

Reviewer 2: ```suggestion
Configuration for Relay is recorded in either the file `.relay/config.yml` or gets overridden through an environment variable. To change the config location, pass the `--config` option to any Relay command:
```

Controls how Relay obtains the project configuration for events. For detailed
explanation of these modes, see [Relay Modes](/product/relay/modes/).

> > enviroment variable: "RELAY_MODE".

Reviewer: I'd put this next to the config name `relay.mode`, but this is probably something the docs team has an opinion on.

Reviewer 2: I'm wondering if it would be possible to turn all the lists on this page into tables (like this: https://docs.sentry.io/platforms/javascript/session-replay/configuration/) - I think it would make it easier to read.

Author: I think that should be a separate PR, could you check out this draft?

https://github.com/getsentry/sentry-docs/pull/7084

Timeout for establishing connections with the upstream in seconds.

> This includes SSL handshakes. Relay reuses connections when the upstream

Author: Since SSL is no longer supported, this sentence is not relevant any longer.

The maximum number of seconds a query is allowed to take across retries.
Individual requests have lower timeouts.

`limits.max_connection_rate`

: _Integer, default: `256`_

The maximum number of connections to Relay that can be created at once.

`limits.max_pending_connections`

: _Integer, default: `2048`_

The maximum number of pending connects to Relay. This corresponds to the
backlog param of `listen(2)` in POSIX.

`limits.max_connections`

: _Integer: default: `25_000`_

The maximum number of open incoming connections to Relay.

> 

Author: Deleted due to the following note in relay release `23.4.0`:

```
Connection config options max_connections, max_pending_connections, and max_connection_rate no longer have an effect. Instead, configure the reverse proxy to handle connection concurrency as needed.
```
